src/lib/hooks.ts

```diff
@@ -29,3 +32,71 @@ export const useNotificationFromLocation = () => {
     }, []);
     return locationState;
 };
+
+type ApiResult<U> = RemoteData.RemoteData<ApiError | undefined, U>;
+// function useDispatchCall<T, U>(
+//     fn: AsyncThunk<U, T, { rejectValue: ApiError }>
+// ): [ApiResult<U>, (args: T, onSuccess?: (result: U) => void) => void] {
+//     const [apiResult, setApiResult] = useState<ApiResult<U>>(RemoteData.initial);
+//     const dispatch = useAppDispatch();
+
+//     const callFn = React.useCallback(
+//         (args: T, onSuccess?: (result: U) => void) => {
+//             if (!RemoteData.isPending(apiResult)) {
+//                 setApiResult(RemoteData.pending);
+
+//                 dispatch(fn(args)).then((action) => {
+//                     if (fn.fulfilled.match(action)) {
+//                         setApiResult(RemoteData.success(action.payload));
+//                         onSuccess?.(action.payload);
+//                     } else {
+//                         setApiResult(RemoteData.failure(action.payload));
+//                     }
+//                 });
+//             }
+//         },
+//         [apiResult]
+//     );
+
+//     return [apiResult, callFn];
+// }
```

Comment:
Kan godt ta med denne i samme slengen.
Et par navngivingsforslag:
```suggestion
export function useAsyncApiActionCreator<T, U>(
    actionCreator: AsyncThunk<U, T, { rejectValue: ApiError }>
): [ApiResult<U>, (args: T, onSuccess?: (result: U) => void) => void] {
    const [apiResult, setApiResult] = useState<ApiResult<U>>(RemoteData.initial);
    const dispatch = useAppDispatch();

    const callFn = React.useCallback(
        (args: T, onSuccess?: (result: U) => void) => {
            if (!RemoteData.isPending(apiResult)) {
                setApiResult(RemoteData.pending);

                dispatch(actionCreator(args)).then((action) => {
                    if (actionCreator.fulfilled.match(action)) {
                        setApiResult(RemoteData.success(action.payload));
                        onSuccess?.(action.payload);
                    } else {
                        setApiResult(RemoteData.failure(action.payload));
                    }
                });
            }
        },
        [apiResult, actionCreator, onSuccess]
    );

    return [apiResult, callFn];
}
```